pkg/model/project.go

```diff
@@ -518,3 +518,33 @@ func (p *Project) DeleteRBACRole(name string) error {
 	}
 	return fmt.Errorf("%s role does nott exist", name)
 }
+
+func (p *ProjectRBACRole) HasPermission(typ ProjectRBACResource_ResourceType, action ProjectRBACPolicy_Action) bool {
```

Comment:
This method is called by [this](https://github.com/pipe-cd/pipecd/pull/3868/files#diff-1c6241b37c87f74eead2ec6ce129dd88bd10cd8ba1ebd5daa2246becd00e0c47R87).